### ISSUE TITLE
Polyfill: then hackery

### ISSUE THREAD

**Issue by domenic** [GoogleChrome/tasklets#15]
I haven't tried to understand the `then` hackery in much detail, but I'm curious how much of it you anticipate making its way to the final spec, versus being expedient for polyfill purposes?

**Comment by surma** [comment]
That is a good question, because I have no idea how much it will have to go into the spec. [This weird thing](https://github.com/GoogleChrome/tasklets/blob/7fac437adac2d7337be555512d36c8daccd9f93a/polyfill/tasklet-polyfill.js#L45) is just in there because otherwise the implicit `then` from an `await` would be passed to the proxy object instead of resolving to the proxy itself. In an older version I was returning the proxy from `get()`, but that caused more problems when handling promises in general.

Bottom line: I hope this is only hack specifically for the polyfill.
